crates/state/src/chain_state.rs

```diff
@@ -51,16 +51,16 @@ pub struct Chainstate {
 // TODO: FIXME: Note that this is used as a temporary solution for the state root calculation
 // It should be replaced once we swap out Chainstate's type definitions with SSZ type definitions
 // which defines all of this more rigorously
-#[derive(BorshSerialize)]
-struct HashedChainstate {
-    last_block: Buf32,
-    slot: u64,
-    epoch: u64,
-    l1_state_hash: Buf32,
-    pending_withdraws_hash: Buf32,
-    exec_env_hash: Buf32,
-    operators_hash: Buf32,
-    deposits_hash: Buf32,
+#[derive(BorshSerialize, BorshDeserialize, Clone, Copy)]
```

Comment:
Maybe this is needed for the proof, but what's this all being added for?  Ideally with SSZ we can completely remove this type.  Should add comment.

Reply:
`Copy` and `Clone` are mainly for passing it around. Was there a reason these weren't already added before?

As for `BorshDeserialize`, it's for the proofs but also for testing. The bridge proof relies on the chain state so some of the unit tests stores the state in a file and reads off it.

Reply:
They weren't added since it was just meant as an implementation detail for the state hashing.